static/app/components/modals/addDashboardWidgetModal.tsx

```diff
@@ -383,7 +384,7 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
       return {...newState, errors: undefined};
     });
 
-    if (field === 'displayType') {
+    if (field === 'displayType' && value !== displayType) {
```

Comment:
I noticed the `setState` above still sends out an analytics event, do you think we should also wrap that with a check against the `prevState`?

I suppose this is an edge case and we won't gain much from either getting those events or choosing to ignore them

Reply:
Thinking about this again, we probably want to keep it because in the small chance that this happens often, it'll give us visibility. Leaving the initial comment in just in case.

Reply:
Good points! Yeah agreed, it can help us identify if someone is intentionally clicking the same visualization type again, so it's probably not a bad idea to keep it.